File: bitlap-core/src/main/kotlin/org/bitlap/core/sql/udf/UDF1.kt
/* Copyright (c) 2022 bitlap.org */
package org.bitlap.core.sql.udf

/**
 * A generic interface for defining user-defined functions.
 *
 * [V]: input value type
 * [R]: result type
 */
interface UDF1<V, R> : UDF {

    /**
     * eval with one input.
     */
    fun eval(input: V): R
}

File: bitlap-core/src/main/scala/org/bitlap/core/sql/udf/UDF1.scala
/**
 * Copyright (C) 2023 bitlap.org .
 */
package org.bitlap.core.sql.udf

/** A generic interface for defining user-defined functions.
 *
 *  [V]: input value type [R]: result type
 */
trait UDF1[V, R] extends UDF {

  /** eval with one input.
   */
  def eval(input: V): R
}
